[Mil_leon] iP

Mil-leon opened the pull request:
# Bebop

Bebop 🤠 is a **FREE** task manager for the you
- Text Based
- Easy to learn
- Fast

All you need to do is 
1. click this [link](https://github.com/Mil-leon/ip/releases)
2. download it
3. add tasks
4. let it handle tasks!

- [x] Manage Todo
- [x] Manage Deadlines
- [x] Manage Events
- [ ] Reminder

You can use it to practice your Java Code

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```
> [!NOTE] 
> Bebop is a work in progress and will be updated gradually

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Mil-leon committed:
Changed the name from Duke to Bebop

Mil-leon committed:
Changed the greetings and the goodbyes

Mil-leon committed:
Add echo feature to the chatbot

Mil-leon committed:
Add Tasklist

Mil-leon committed:
Add Task class and Mark/Unmark features

Mil-leon committed:
Add ToDo, Deadline, Event classes and implemented their functions

Mil-leon committed:
Add Testcases and Expected

Mil-leon committed:
String Validation for inputs and created BebopException

Mil-leon committed:
Completed delete function

Mil-leon committed:
Fix to follow coding standards

Mil-leon committed:
Create Bebop.txt to save data of task

Mil-leon committed:
Change the format of BebopException

Mil-leon committed:
Implement save feature for Bebop and edit task classes

Mil-leon committed:
Merge branch 'branch-level-7'

Mil-leon committed:
Create introduction function for Bebop

Mil-leon committed:
Add LocalDate to Deadline and Event class

Mil-leon committed:
Add LocalDateTime and error detection for wrong format

Mil-leon committed:
Merge branch 'branch-level-8'

Mil-leon committed:
Add Ui, Command, Parser, Storage, TaskList, Ui Classes

Mil-leon committed:
Add error detection for delete and mark commands

Mil-leon committed:
Merge branch 'more-oop'

Mil-leon committed:
Add packages Command, Tasks, Ui, Exceptions

Mil-leon committed:
Merge branch 'add-gradle-support'

Mil-leon committed:
Fix Delete and Mark Command Error

Mil-leon committed:
Add AddCommandTest and DeleteCommandTest

Mil-leon committed:
Merge branch 'A-JUnit'

Mil-leon committed:
Chang shadowJar to create Fat Jar

Mil-leon committed:
Merge branch 'A-Jar'

Mil-leon committed:
Add all Javadoc into the class/methods

Mil-leon committed:
Add Find Command

Mil-leon committed:
Merge branch 'branch-Level-9'

Mil-leon committed:
Merge branch 'branch-A-JavaDoc'

Mil-leon committed:
Add checkstyle + suppression xml files

Mil-leon committed:
Fix according to check-style standards

Mil-leon committed:
Merge branch 'A-Checkstyle'

broccoli-hater commented on src/main/java/bebop/command/AddCommand.java:
```diff
@@ -0,0 +1,155 @@
+package bebop.command;
+import java.time.LocalDateTime;
```
The import statements are well-structured and formatted correctly! Although there should be a new line after the package before the first import. I noticed this issue in some other classes too

Mil-leon replied:
Thanks for the feedback :), will update the mistakes accordingly.

broccoli-hater commented on src/main/java/bebop/command/DeleteCommand.java:
```diff
@@ -0,0 +1,46 @@
+     */
+    @Override
+    public boolean execute(TaskList tasks, Ui ui, Storage storage) throws BebopException {
+        String[] input = this.command.split(" ");
```
Do consider making the String array `input` to be plural.

broccoli-hater commented on src/main/java/bebop/command/AddCommand.java:
```diff
@@ -0,0 +1,155 @@
+     */
+    @Override
+    public boolean execute(TaskList tasks, Ui ui, Storage storage) throws BebopException {
+        if (this.type.equals("t")) {
```
It's great that you've made separate classes for the different commands? Nice job!
I would recommend you to further subclass the different types of 'AddCommand' classes.

broccoli-hater commented on src/main/java/bebop/command/AddCommand.java:
```diff
@@ -0,0 +1,155 @@
+    public boolean execute(TaskList tasks, Ui ui, Storage storage) throws BebopException {
+        if (this.type.equals("t")) {
+            String[] todos = command.split("todo ");
+            if (isFormatted(todos, "t")) {
```
Although this way works too, consider putting the relevant input parsing code into the 'Parser' class

broccoli-hater commented on src/main/java/bebop/task/Deadline.java:
```diff
@@ -0,0 +1,58 @@
+
+public class Deadline extends Task {
+    private final String start;
+    private final LocalDateTime startDate;
```
Maybe you could change the name for the 'LocalDateTime' variable as startDate does not make much sense if the deadline is to be completed by then.

broccoli-hater commented on src/main/java/bebop/ui/Ui.java:
```diff
@@ -0,0 +1,61 @@
+        System.out.println("\tTODO : todo EVENTNAME");
+        System.out.println("\tDEADLINE : deadline EVENTNAME /by ENDTIME");
+        System.out.println("\tEVENT : event EVENTNAME /from STARTTIME /to ENDTIME");
+    }
```
Nice graphic!

broccoli-hater commented on src/main/java/bebop/ui/Ui.java:
```diff
@@ -0,0 +1,61 @@
+    /**
+     * adds a divider between commands.
+     */
+    public void divider() {
```
Consider changing the name of the method to a verb! For example, 'printDivider()'

broccoli-hater reviewed (commented):
Overall, the code quality is great, the namings are intuitive and the coding standard is adhered to except in some minor cases.

broccoli-hater commented on src/test/java/bebop/command/DeleteCommandTest.java:
```diff
@@ -0,0 +1,37 @@
+
+public class DeleteCommandTest {
+    @Test
+    public void DeleteCommandTestSuccess() throws BebopException {
```
While normal method names are written in camelCase, test methods usually follow another convention:
'unitBeingTested_descriptionOfTestInputs_expectedOutcome'

Mil-leon committed:
Add JavaFx and UI

Mil-leon committed:
Add all the GUI format for the commands

Mil-leon committed:
Delete the run method

The run method was in charge of the old terminal ui.
With the addition of a GUI it is no longer necessary.

Mil-leon committed:
Add assertions

Assertions to check for task state + input length have been added

Mil-leon committed:
Shorten the execute command by adding new methods

Mil-leon committed:
Merge branch 'branch-A-Streams' into branch-A-Assertions

Mil-leon committed:
Add Github CI

Mil-leon committed:
Fix the find command and add bettersearch

Mil-leon committed:
Merge branch 'c-BetterSearch'

Mil-leon committed:
Merge pull request #1 from Mil-leon/branch-A-CodeQuality

Shorten the execute command by adding new methods

Mil-leon committed:
Add assertions to add command

Mil-leon committed:
Merge branch 'branch-A-Assertions'

Mil-leon committed:
Delete a comment leading to a bug

Mil-leon committed:
Fix errors from previous weeks

Mil-leon committed:
Add new error handling

Mil-leon committed:
Add new JUnit test case

Mil-leon committed:
Merge branch 'branch-A-MoreTesting'

Mil-leon committed:
Made the GUI better and added new images

Mil-leon committed:
Merge branch 'branch-A-BetterGui'

Mil-leon committed:
Change the photos used for duke and user

Mil-leon committed:
Fix the checkstyle and more javadocs added

Mil-leon committed:
Update README.md

Mil-leon committed:
Add Ui.png and cleared bebop.txt

Mil-leon committed:
Update README.md

Mil-leon committed:
Update tasklist

Mil-leon committed:
Merge branch 'master' of https://github.com/Mil-leon/ip

Mil-leon committed:
Update README.md

Mil-leon committed:
Final fix of JavaDocs

Mil-leon committed:
Merge branch 'master' of https://github.com/Mil-leon/ip

Mil-leon committed:
Change to java 17
